src/Aspire.Dashboard/Components/Pages/ConsoleLogs.razor.cs

```diff
@@ -21,62 +21,69 @@ public partial class ConsoleLogs : ComponentBase, IAsyncDisposable
     [Parameter]
     public string? ResourceName { get; set; }
 
-    private bool ConvertTimestampsFromUtc => _selectedResource is ContainerViewModel;
+    private readonly TaskCompletionSource _whenFirstRenderComplete = new();
+    private readonly CancellationTokenSource _resourceSubscriptionCancellation = new();
+    private readonly CancellationSeries _logSubscriptionCancellationSeries = new();
+    private readonly Dictionary<string, ResourceViewModel> _resourceByName = [];
 
+    // UI
     private FluentSelect<Option<string>>? _resourceSelectComponent;
+    private Option<string> _noSelection = null!;
+    private LogViewer _logViewer = null!;
+
+    // State
+    private string _status = null!;
     private Option<string>? _selectedOption;
     private ResourceViewModel? _selectedResource;
-    private readonly Dictionary<string, ResourceViewModel> _resourceNameMapping = new();
-    private List<Option<string>>? Resources { get; set; }
-    private LogViewer? _logViewer;
-    private readonly CancellationTokenSource _watchResourcesCts = new();
-    private CancellationTokenSource? _watchLogsTokenSource;
-    private string _status = "...";
+    private bool? _initialisedSuccessfully;
 
-    private readonly TaskCompletionSource _renderCompleteTcs = new();
-
-    private Option<string> _noSelection = null!;
+    private List<Option<string>>? _resources;
 
     protected override void OnInitialized()
     {
         _noSelection = new() { Value = null, Text = Loc[Dashboard.Resources.ConsoleLogs.ConsoleLogsSelectAResource] };
         _status = Loc[Dashboard.Resources.ConsoleLogs.ConsoleLogsLoadingResources];
 
-        var (snapshot, subscription) = ResourceService.Subscribe();
+        TrackResources();
 
-        foreach (var resource in snapshot)
-        {
-            _resourceNameMapping[resource.Name] = resource;
-        }
-
-        UpdateResourcesList();
+        StateHasChanged();
```

Comment:
Is this needed?

Reply:
No, I'll remove it. This was in the old code too, but I agree it seems like it shouldn't be needed, and testing without this works fine too.

Reply:
That was originally in there when this code was in OnAfterRenderAsync - it was needed at the time (or, probably more accurately, it was a hacky workaround to something at the time) - but when we moved all this code to OnInitialized it likely became irrelevant.